// vuestic-ui/dist/es/src/components/va-file-upload/VaFileUploadList/VaFileUploadList.vue_vue_type_script_lang-68513a34.mjs
import { defineComponent, computed } from "vue";
import { V as VaFileUploadListItem } from "../VaFileUploadListItem/index-49933912.mjs";
import { V as VaFileUploadGalleryItem } from "../VaFileUploadGalleryItem/index-b60f1d54.mjs";
import { V as VaFileUploadSingleItem } from "../VaFileUploadSingleItem/index-9708a40e.mjs";
import { b as VaList } from "../../va-list/index-8c74adbc.mjs";
import { e as extractComponentProps } from "../../../utils/component-options/extract-component-options-6b96d215.mjs";
import { f as filterComponentProps } from "../../../utils/component-options/filter-props-9727421a.mjs";
const VaFileUploadGalleryItemProps = extractComponentProps(VaFileUploadGalleryItem);
const VaFileUploadListItemProps = extractComponentProps(VaFileUploadListItem);
const VaFileUploadSingleItemProps = extractComponentProps(VaFileUploadSingleItem);
const _sfc_main = defineComponent({
  name: "VaFileUploadList",
  components: {
    VaList,
    VaFileUploadListItem,
    VaFileUploadGalleryItem,
    VaFileUploadSingleItem
  },
  emits: ["remove", "removeSingle"],
  props: {
    type: { type: String, default: "" },
    files: { type: Array, default: null },
    ...VaFileUploadGalleryItemProps,
    ...VaFileUploadListItemProps,
    ...VaFileUploadSingleItemProps
  },
  setup(props) {
    const filesList = computed(() => props.files.map(convertFile));
    const convertFile = (file) => ({
      name: file.name || file.url || "",
      size: formatSize(file.size),
      date: formatDate(/* @__PURE__ */ new Date()),
      image: file
    });
    const formatSize = (bytes) => {
      if (bytes === 0) {
        return "0 Bytes";
      }
      if (!bytes) {
        return "";
      }
      const k = 1024;
      const sizes = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
      const i = Math.floor(Math.log(bytes) / Math.log(k));
      return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + " " + sizes[i];
    };
    const formatDate = (date = /* @__PURE__ */ new Date()) => {
      return date.toLocaleDateString("en-US", {
        hour: "2-digit",
        minute: "2-digit",
        month: "short",
        day: "numeric",
        year: "numeric"
      });
    };
    return {
      galleryItemProps: filterComponentProps(VaFileUploadGalleryItemProps),
      itemProps: filterComponentProps(VaFileUploadListItemProps),
      singleItemProps: filterComponentProps(VaFileUploadSingleItemProps),
      filesList
    };
  }
});
export {
  _sfc_main as _
};

import '../../../../VaFileUploadList.css';
